diff --git a/src/rabbit_disk_monitor.erl b/src/rabbit_disk_monitor.erl
index 68f6095..da35ef0 100644
--- a/src/rabbit_disk_monitor.erl
+++ b/src/rabbit_disk_monitor.erl
@@ -205,7 +205,7 @@ get_disk_free(Dir, {unix, Sun})
   when Sun =:= sunos; Sun =:= sunos4; Sun =:= solaris ->
     parse_free_unix(rabbit_misc:os_cmd("/usr/bin/df -k " ++ Dir));
 get_disk_free(Dir, {unix, _}) ->
-    parse_free_unix(rabbit_misc:os_cmd("/bin/df -kP " ++ Dir));
+    parse_free_unix(rabbit_misc:os_cmd("%coreutils%/bin/df -kP " ++ Dir));
 get_disk_free(Dir, {win32, _}) ->
     parse_free_win32(rabbit_misc:os_cmd("dir /-C /W \"" ++ Dir ++ "\"")).
 
